bands/site_sections/bands.py

```diff
@@ -9,57 +9,49 @@ def index(self, session, id, message=''):
             'band': session.band(id)
         }
 
-    def agreement(self, session, message='', **params):
-        band = session.band(params, restricted=True)
+    def agreement(self, session, band_id, message='', **params):
+        band_info = session.band_info(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.performer_count:
+            if not band_info.performer_count:
                 message = 'You must tell us how many people are in your band'
-            elif not band.poc_phone:
+            elif not band_info.poc_phone:
                 message = 'You must enter an on-site point-of-contact cellphone number'
-            elif not band.arrival_time:
+            elif not band_info.arrival_time:
                 message = 'You must enter your expected arrival time'
-            elif band.bringing_vehicle and not band.vehicle_info:
+            elif band_info.bringing_vehicle and not band.info.vehicle_info:
                 message = 'You must provide your vehicle information'
             else:
-                raise HTTPRedirect('index?id={}&message={}', band.id, 'Your band information has been uploaded')
+                raise HTTPRedirect('index?id={}&message={}', band_info.band_id, 'Your band information has been uploaded')
 
         return {
-            'band': band,
+            'band': band_info,
             'message': message
         }
 
-    def bio(self, session, message='', bio_pic=None, **params):
-        band = session.band(params, restricted=True)
+    def bio(self, session, band_id, message='', bio_pic=None, **params):
+        band_bio = session.band_bio(params, restricted=True)
+        band_bio.band_id = band_id
         if cherrypy.request.method == 'POST':
-            if not band.bio:
+            if not band_bio.bio:
                 message = 'Please provide a brief bio for our website'
 
-            if not message and bio_pic:
```

Comment:
Whoops, this was taken out during testing because for some reason I kept getting the 'extension' error message even though the bio pic file picker was blank.